### PR TITLE
Fix issue #182

### PR DESCRIPTION
**Pull request by CermakM**
Issue [#182](https://github.com/thoth-station/thamos/issues/182):

```
__init__() got an unexpected keyword argument 'recommendation_type'
```

Signed-off-by: Marek Cermak <macermak@redhat.com>

modified:   thamos/lib.py

### CODE REVIEW

File: thamos/lib.py [hunk 1]
```diff
@@ -183,15 +183,17 @@ def advise(
...
         limit_latest_versions = thoth_config.content.get("limit_latest_versions")
 
     if recommendation_type is None:
-        recommendation_type = thoth_config.content.get("recommendation_type") or "stable"
-
-    if runtime_environment is None:
-        runtime_environment = thoth_config.get_runtime_environment(runtime_environment_name)
+        recommendation_type = thoth_config.content.get(
+            "recommendation_type"
+        ) or runtime_environment.pop("recommendation_type", "stable")
```
Comment (fridex) [inline thamos/lib.py]: I think pop'ing `recommendation_type` will need to be done always to preserve case when it stays in the `runtime_envrionment` dict (retrieved from config file and `recommendation_type` is set on this function call). So something like (after this if) should fix that:

```python
runtime_environment.pop("recommendation_type", None)
```

Comment (CermakM) [reply thamos/lib.py]: Ah, I see, so how about reverting and on line 205:

```
if "recommendation_type" in runtime_environment and recommendation_type is None:
```
we just got rid of the `and` part of the condition -- the `recommendation_type` will never be `None`?

Comment (fridex) [reply thamos/lib.py]: Sounds like another approach :+1:

Comment (CermakM) [reply thamos/lib.py]: Cool, thanks!

**Pull request merged**